2.1: Cache being written to temp directory

**Describe the bug**

```
SimpleSAML\Error\Error: UNHANDLEDEXCEPTION
Backtrace:
1 public/_include.php:28 (SimpleSAML_exception_handler)
0 [builtin] (N/A)
Caused by: RuntimeException: Unable to write in the "cache" directory (/tmp/simplesaml/cache/saml).
Backtrace:
5 vendor/symfony/http-kernel/Kernel.php:641 (Symfony\Component\HttpKernel\Kernel::buildContainer)
4 vendor/symfony/http-kernel/Kernel.php:545 (Symfony\Component\HttpKernel\Kernel::initializeContainer)
3 vendor/symfony/http-kernel/Kernel.php:787 (Symfony\Component\HttpKernel\Kernel::preBoot)
2 vendor/symfony/http-kernel/Kernel.php:190 (Symfony\Component\HttpKernel\Kernel::handle)
1 src/SimpleSAML/Module.php:234 (SimpleSAML\Module::process)
0 public/module.php:17 (N/A)
```

The /tmp directory is being emptied on most OS'es, so we should be writing Symfony cache elsewhere (like /var/cache)

--- a/config/config.php.dist
+++ b/config/config.php.dist
@@ -58,6 +58,7 @@ $config = [
     /*
      * The following settings are *filesystem paths* which define where
      * SimpleSAMLphp can find or write the following things:
+     * - 'cachedir': Where Symfony can write it's cache. 
      * - 'loggingdir': Where to write logs. MUST be set to NULL when using a logging
      *                 handler other than `file`.
      * - 'datadir': Storage of general data.
@@ -66,9 +67,10 @@ $config = [
      * When specified as a relative path, this is relative to the SimpleSAMLphp
      * root directory.
      */
+    'cachedir' => '/var/cache/simplesamlphp',
     //'loggingdir' => '/var/log/',
     //'datadir' => '/var/data/',
-    'tempdir' => '/tmp/simplesaml',
+    'tempdir' => '/tmp/simplesamlphp',
 
     /*
      * Certificate and key material can be loaded from different possible

--- a/config/config.php.dist
+++ b/config/config.php.dist
@@ -58,7 +58,7 @@ $config = [
     /*
      * The following settings are *filesystem paths* which define where
      * SimpleSAMLphp can find or write the following things:
-     * - 'cachedir': Where Symfony can write it's cache. 
+     * - 'cachedir': Where SimpleSAMLphp can write its cache. 
      * - 'loggingdir': Where to write logs. MUST be set to NULL when using a logging
      *                 handler other than `file`.
      * - 'datadir': Storage of general data.

--- a/config/config.php.dist
+++ b/config/config.php.dist
@@ -62,7 +62,6 @@ $config = [
      * - 'loggingdir': Where to write logs. MUST be set to NULL when using a logging
      *                 handler other than `file`.
      * - 'datadir': Storage of general data.
-     * - 'tempdir': Saving temporary files. SimpleSAMLphp will attempt to create
      *   this directory if it doesn't exist.
      * When specified as a relative path, this is relative to the SimpleSAMLphp
      * root directory.
@@ -70,7 +69,6 @@ $config = [
     'cachedir' => '/var/cache/simplesamlphp',
     //'loggingdir' => '/var/log/',
     //'datadir' => '/var/data/',
-    'tempdir' => '/tmp/simplesamlphp',
 
     /*
      * Certificate and key material can be loaded from different possible

--- a/src/SimpleSAML/Kernel.php
+++ b/src/SimpleSAML/Kernel.php
@@ -51,8 +51,7 @@ class Kernel extends BaseKernel
     public function getCacheDir(): string
     {
         $configuration = Configuration::getInstance();
-        $cachePath = $configuration->getString('tempdir') . DIRECTORY_SEPARATOR
-            . 'cache' . DIRECTORY_SEPARATOR . $this->module;
+        $cachePath = $configuration->getString('cachedir') . DIRECTORY_SEPARATOR . $this->module;
 
         $sysUtils = new System();
         if ($sysUtils->isAbsolutePath($cachePath)) {

